stress/common/src/main/java/alluxio/stress/client/ClientIOTaskResult.java

```diff
@@ -0,0 +1,351 @@
+
+      // Iterate over all operations
+      for (ClientIOOperation operation : ClientIOOperation.values()) {
+        for (Boolean readRandom : Arrays.asList(false, true)) {
+          List<ClientIOTaskResult> opSummaries =
+              summaries.stream().filter(x -> x.mParameters.mOperation == operation)
+                  .filter(x -> x.mParameters.mReadRandom == readRandom)
+                  .collect(Collectors.toList());
+
+          if (!opSummaries.isEmpty()) {
+            // first() is the list of common field names, second() is the list of unique field names
+            Pair<List<String>, List<String>> fieldNames = Parameters.partitionFieldNames(
+                opSummaries.stream().map(x -> x.mParameters).collect(Collectors.toList()));
+
+            // Split up common description into 100 character chunks, for the sub title
+            List<String> subTitle = new ArrayList<>(Splitter.fixedLength(100).splitToList(
+                opSummaries.get(0).mParameters.getDescription(fieldNames.getFirst())));
+
+            for (ClientIOTaskResult summary : opSummaries) {
+              String series = summary.mParameters.getDescription(fieldNames.getSecond());
+              subTitle.add(series + ": " + DateFormat.getDateTimeInstance()
+                  .format(summary.computeLastEndMs()));
+            }
+
+            LineGraph responseTimeGraph = new LineGraph(
+                operation + (readRandom ? " - Random" : " - Sequential") + " - Throughput",
+                subTitle, "# Threads", "Throughput (MB/s)");
+
+            for (ClientIOTaskResult summary : opSummaries) {
+              String series = summary.mParameters.getDescription(fieldNames.getSecond());
+              responseTimeGraph.addDataSeries(series, summary.getThroughputData());
+              responseTimeGraph.setErrors(series, summary.collectErrors());
+            }
+
+            graphs.add(responseTimeGraph);
+          }
+        }
+      }
+
+      return graphs;
+    }
+  }
+
+  /**
+   * A result for a single thread count test.
+   */
+  public static final class ThreadCountResult {
+    private long mRecordStartMs;
+    private long mEndMs;
+    private long mIOBytes;
+    private List<String> mErrors;
+
+    /**
+     * Creates an instance.
+     */
+    public ThreadCountResult() {
+      // Default constructor required for json deserialization
+      mErrors = new ArrayList<>();
+    }
+
+    /**
+     * Merges (updates) a result with this result.
+     *
+     * @param result  the result to merge
+     */
+    public void merge(ClientIOTaskResult.ThreadCountResult result) {
+      mRecordStartMs = Math.min(mRecordStartMs, result.mRecordStartMs);
+      mEndMs = Math.max(mEndMs, result.mEndMs);
+      mIOBytes += result.mIOBytes;
+      mErrors.addAll(result.mErrors);
+      update();
+    }
+
+    /**
+     * Update various measurements given the existing measurements.
+     */
+    public void update() {
+    }
```

Comment:
was this a copy from another class. since it's `ThreadCountResult` type and not bound by any inherited classes or interfaces can this just be removed? Same for the other methods in this class that are empty.

Reply:
oops, removed. I think the other ones which are setters need to be there for jackson json processing.